llvm/test/CodeGen/AMDGPU/srem.ll

```diff
@@ -4861,625 +4861,610 @@ define amdgpu_kernel void @srem_v4i64(ptr addrspace(1) %out, ptr addrspace(1) %i
 ; GCN-LABEL: srem_v4i64:
 ; GCN:       ; %bb.0:
 ; GCN-NEXT:    s_load_dwordx4 s[4:7], s[2:3], 0x24
-; GCN-NEXT:    v_mov_b32_e32 v8, 0
+; GCN-NEXT:    v_mov_b32_e32 v4, 0
```

Comment:
The PR has reduced the lines of code by 15, and reduced VGPR usage from 25->18. That said, I'm still investigating the changes here.